message MessageEvent {
    required string sys_id = 1;
    repeated Event event = 2;

    message Event {
        required EventType type = 1 [default = BASIC];
        required string name = 2;
        required double timestamp = 3;
        repeated Argument arg = 4;
    }

    message Argument {
        optional string name = 1;
        required string value = 2;
    }

    enum EventType {
        BASIC = 0;
        SECTION = 1;
        COUNTER = 2;
    }
}
